### ISSUE TITLE
Implement Collection Selection Dialog

### ISSUE THREAD

**Issue by jmchilton** [galaxyproject/galaxy#9114]
This is the collection issue mirroring the data issue outlined in https://github.com/galaxyproject/galaxy/issues/9110. The idea outlined there (and in http://bit.ly/simplified-workflow-ui) is that we can simplify the UX for running workflows by eliminating the tool panel and upload options if we can provide context-aware alternatives directly from the workflow run form. #8512 introduces a new collection selection dialog component, it needs to be enhanced significantly as follows:

- [ ] Add a button to the tool form for this widget to match the button for dataset.
- [ ] Delegate to the upload component to allow upload/creation of collections directly from the selection component (or a button next to it on the tool form).
- [ ] Allow restricting the upload and create options based on the collection type annotated in the tool form.
- [ ] Allow restricting the upload and create options based on the new format for the collection that will be annotated after #9086.

Stretch Goals:

 - [ ] Allow selecting collections from other histories.
 - [ ] Allow selecting library folders and creating collections from them.
